src/mlpack/core/cv/metrics/roc_auc_score_impl.hpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * @file core/cv/metrics/roc_auc_score_impl.hpp
+ * @author Sri Madhan M
+ *
+ * Implementation of the area under Receiver Operating Characteristic curve
+ * (ROC-AUC) score.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_METRICS_ROCAUCSCORE_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_ROCAUCSCORE_IMPL_HPP
+
+#include <mlpack/core/cv/metrics/accuracy.hpp>
+#include <iostream>
+
+namespace mlpack {
+namespace cv {
+
+template<size_t PC /* PositiveClass */>
```

Comment:
```suggestion
template<size_t PositiveClass>
```

It's pedantic but I would just use `PositiveClass` throughout.